packages/core/src/internal/utils/identity.ts

```diff
@@ -81,3 +82,43 @@ export function deepClone(obj: any) {
 
   return isMap(obj) ? cloneMap(obj) : JSON.parse(JSON.stringify(obj));
 }
+
+export function anyOrAllPropertiesPass(obj: any, propValuePairs: string, anyOrAll: 'any' | 'all'): boolean {
+  if (!propValuePairs) {
+    return true;
+  }
+
+  const tests = convertStringPropValuePairsToTuple(propValuePairs);
+
+  if (!obj) {
+    return false;
+  }
+
+  if (tests.length < 1) {
+    return true;
+  } else {
+    const testResults = tests.filter(pvArry => {
+      const [propname, expectedVal] = pvArry;
+
+      if (expectedVal === false) {
+        return !obj[propname];
+      }
+
+      return obj[propname] === expectedVal;
+    });
+
+    return anyOrAll === 'all' ? testResults.length === tests.length : testResults.length > 0;
+  }
+}
+
+export function allPropertiesPass(obj: any, propValuePairs: string): boolean {
+  return anyOrAllPropertiesPass(obj, propValuePairs, 'all');
+}
+
+export function anyPropertiesPass(obj: any, propValuePairs: string): boolean {
+  return anyOrAllPropertiesPass(obj, propValuePairs, 'any');
+}
```

Comment:
These utility functions pave the way for us to get more mileage out of the property value keys I'm using in the web animations api extensions. We could see more utility out of the `propertyName:testValue` style of checks in our code.